### PR TITLE
Cloud nine

### PR DESCRIPTION
**Pull request by EnricoSandri**

### CODE REVIEW

File: Assets/Prefabs/Manager/ManagerGameObject/Manager.cs [hunk 1]
```diff
@@ -258,14 +261,16 @@ public bool TryToBuild()
         {
             GameObject toBeCloned = placeHolderPrefabs[currentSelection];
             GameObject newBuilding = GameObject.Instantiate(toBeCloned, newPlaceIndicator.transform.position, newPlaceIndicator.transform.rotation);
-
-           // newBuilding.transform.localScale = new Vector3(newBuilding.transform.localScale.x, placeIndicator.transform.localScale.y * newBuilding.transform.localScale.y, newBuilding.transform.localScale.z);
+            // newBuilding.transform.localScale = new Vector3(newBuilding.transform.localScale.x, placeIndicator.transform.localScale.y * newBuilding.transform.localScale.y, newBuilding.transform.localScale.z);
 
             newBuilding.transform.localScale = new Vector3(newBuilding.transform.localScale.x, newPlaceIndicator.transform.localScale.y * newBuilding.transform.localScale.y, newBuilding.transform.localScale.z);
             //  Material newMat = new Material(newBuilding.GetComponent<MeshRenderer>().material);
             //  newBuilding.GetComponent<MeshRenderer>().material = newMat;
```
Comment (Banananaman91) [inline Assets/Prefabs/Manager/ManagerGameObject/Manager.cs:271]: Remove any unused code

File: Assets/Prefabs/Manager/ManagerGameObject/Manager.cs [hunk 2]
```diff
@@ -258,14 +261,16 @@ public bool TryToBuild()
...
 
             newBuilding.transform.localScale = new Vector3(newBuilding.transform.localScale.x, newPlaceIndicator.transform.localScale.y * newBuilding.transform.localScale.y, newBuilding.transform.localScale.z);
             //  Material newMat = new Material(newBuilding.GetComponent<MeshRenderer>().material);
             //  newBuilding.GetComponent<MeshRenderer>().material = newMat;
             //Note for Jack. Use one material.
-            virtualCurrency.SealsCurrency -= allResources.resourceInfo[unlockedBuildOptionsList[currentSelection]].buildingCost;
+
+            //USING THE "toBeCloned" NAME AS A STRING FOR THE ITEM PUCHASE ON CLOUD
+            virtualCurrency.PurchaseUpgrade(toBeCloned.name, allResources.resourceInfo[unlockedBuildOptionsList[currentSelection]].buildingCost);
+            // virtualCurrency.SealsCurrency -= allResources.resourceInfo[unlockedBuildOptionsList[currentSelection]].buildingCost; // Removed as PurchaseUpgrade does that.
```
Comment (Banananaman91) [inline Assets/Prefabs/Manager/ManagerGameObject/Manager.cs]: More unused code

File: Assets/Prefabs/Manager/ManagerGameObject/Manager.cs [hunk 3]
```diff
@@ -277,8 +282,13 @@ public void Update()
     {
         if (started)
```
Comment (Banananaman91) [inline Assets/Prefabs/Manager/ManagerGameObject/Manager.cs:284]: Can change to
`if (!started) return;`
to reduce nesting

File: Assets/Prefabs/Manager/ManagerGameObject/Manager.cs [hunk 4]
```diff
@@ -277,8 +282,13 @@ public void Update()
     {
         if (started)
         {
-
             moneyText.text = virtualCurrency.SealsCurrency.ToString();
+            if (Input.GetKeyDown(KeyCode.F5))
+            {
+                virtualCurrency.AddCurrency();
+            }
```
Comment (Banananaman91) [inline Assets/Prefabs/Manager/ManagerGameObject/Manager.cs]: Braces not necessary, can just single line this

File: Assets/Scripts/Enrico/GetPlayerStats.cs [hunk 5]
```diff
@@ -89,8 +89,32 @@ public void FetchCurrency()
...
+    {
+        //PlayFabClientAPI.LoginWithPlayFab(PFA.loginRequest, result => {
+
+        //    virtualCurrency.SealsCurrency = result.InfoResultPayload.UserVirtualCurrency["SL"]; // fetch the currency value.
+        //    gotCurrency = true;
+
+        //}, error => {
+        //    Debug.Log("Error ritriving you currency");
+
+        //}, null);
```
Comment (Banananaman91) [inline Assets/Scripts/Enrico/GetPlayerStats.cs:101]: More unused code that needs removing

File: Assets/Scripts/Enrico/Inventory.cs [hunk 6]
```diff
@@ -1,66 +1,66 @@
...
+    private Manager manager;
+    Resources allResources;
+
+    public void GetCatalog()
+    {
+        manager = FindObjectOfType<Manager>().GetComponent<Manager>();
+        allResources = manager.allResources;
         foreach(var res in allResources.resourceInfo)
         {
             print(res.resourceName);
```
Comment (Banananaman91) [inline Assets/Scripts/Enrico/Inventory.cs]: Could easily be
`Each(allResources.resourceInfo, res => print(res.resourceName));`

File: Assets/Scripts/Enrico/Inventory.cs [hunk 7]
```diff
@@ -1,66 +1,66 @@
...
+                ///////JSON STUFF HERE/////////////////////////////
+                string json = i.CustomData;                                                 // Store cloudData into a string
+                File.WriteAllText(Application.dataPath + "/savedJson.json", json);         //Write a json file to assets
+                string read = File.ReadAllText(Application.dataPath + "/savedJson.json"); //Read from json file and store it
+                JsonDataHolder jsonSet = JsonUtility.FromJson<JsonDataHolder>(read);     // Apply the read data and set the data the json class.
+                ////////////////////////////////////////////////
+                
+                foreach (var info in allResources.resourceInfo)
+                {
+                    if (info.resourceName == i.DisplayName)
```
Comment (Banananaman91) [inline Assets/Scripts/Enrico/Inventory.cs]: Can invert to reduce nesting

File: Assets/Scripts/Enrico/Inventory.cs [hunk 8]
```diff
@@ -1,66 +1,66 @@
...
+                    {
+                        info.buildingCost = (int)cost;
+                        
+                        ////Apply Json Values to the player.
+                        if(jsonSet != null)
+                        {
+                            info.damage = jsonSet.Damage;
+                        }
```
Comment (Banananaman91) [inline Assets/Scripts/Enrico/Inventory.cs]: Can single line this

File: Assets/Scripts/Enrico/VirtualCurrency.cs [hunk 9]
```diff
@@ -36,42 +38,113 @@ public void PurchaseUpgrade(string itemID, int price)
...
+        //   //     else
+        //   //     {
+        //   //         hasItem = false;
+        //   //         Debug.Log(hasItem);
+        //   //     }
+        //   // }
+        //   //// if (hasItem == false)
+        //   // {
+        //    //}
+        //}, error => { Debug.Log(error.ErrorMessage); });
```
Comment (Banananaman91) [inline Assets/Scripts/Enrico/VirtualCurrency.cs]: Lots of unused code that needs removing

File: Assets/Prefabs/Manager/ManagerGameObject/Manager.cs (hunk 2 above)
Comment (EnricoSandri) [reply Assets/Prefabs/Manager/ManagerGameObject/Manager.cs]: OK, good point

File: Assets/Prefabs/Manager/ManagerGameObject/Manager.cs (hunk 3 above)
Comment (EnricoSandri) [reply Assets/Prefabs/Manager/ManagerGameObject/Manager.cs:284]: Not mine

File: Assets/Prefabs/Manager/ManagerGameObject/Manager.cs (hunk 4 above)
Comment (EnricoSandri) [reply Assets/Prefabs/Manager/ManagerGameObject/Manager.cs]: Good point

File: Assets/Scripts/Enrico/GetPlayerStats.cs (hunk 5 above)
Comment (EnricoSandri) [reply Assets/Scripts/Enrico/GetPlayerStats.cs:101]: Will do

File: Assets/Scripts/Enrico/Inventory.cs (hunk 6 above)
Comment (EnricoSandri) [reply Assets/Scripts/Enrico/Inventory.cs]: not mine

File: Assets/Scripts/Enrico/Inventory.cs (hunk 7 above)
Comment (EnricoSandri) [reply Assets/Scripts/Enrico/Inventory.cs]: Good point

File: Assets/Scripts/Enrico/Inventory.cs (hunk 8 above)
Comment (EnricoSandri) [reply Assets/Scripts/Enrico/Inventory.cs]: True, left cos we need to add more. true tho

File: Assets/Scripts/Enrico/VirtualCurrency.cs (hunk 9 above)
Comment (EnricoSandri) [reply Assets/Scripts/Enrico/VirtualCurrency.cs]: need that for tomo

**Review by Banananaman91 — APPROVED** [review]

**Pull request merged**